include/math/utils/ndop.h

```diff
@@ -0,0 +1,221 @@
+#ifndef MATH_NDOP_H_
+#define MATH_NDOP_H_
+
+#include <algorithm>
+#include <array>
+#include <limits>
+#include <numeric>
+#include <numbers>
+#include <functional>
+#include <optional>
+#include <vector>
+
+#include "math/math.h"
+
+namespace math::utils {
+
+struct DOP14_Traits
+{
+    constexpr static size_t get_num_axises() noexcept { return std::size(axises); }
+    constexpr static size_t get_num_planes() noexcept { return get_num_axises() * 2; }
+
+    constexpr size_t get_inverse_plane_index(size_t index) const noexcept {
```

Comment:
I think that's confusing that we work with indecies. That's hard to tell by look which `index` corresponds to what. I suggest giving each axis/plane a label and work with labels instead.